src/hooks/useDelayAnimation.ts

```diff
@@ -0,0 +1,22 @@
+import { useState } from 'react';
+
+export const useDelayAnimation = (delay: number) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isAnimation, setIsAnimation] = useState(false);
+
+  const setOpen = (open?: boolean) => {
+    if (open === undefined) open = !isOpen;
+
```

Comment:
I think this would be clearer : 

 open = open === undefined ? !isOpen : open;